sdks/python/apache_beam/ml/inference/onnx_inference.py

```diff
@@ -0,0 +1,144 @@
+  return ort_session
+
+
+def _convert_to_result(
+    batch: Iterable, predictions: Union[Iterable, Dict[Any, Iterable]]
+) -> Iterable[PredictionResult]:
+  if isinstance(predictions, dict):
+    # Go from one dictionary of type: {key_type1: Iterable<val_type1>,
+    # key_type2: Iterable<val_type2>, ...} where each Iterable is of
+    # length batch_size, to a list of dictionaries:
+    # [{key_type1: value_type1, key_type2: value_type2}]
+    predictions_per_tensor = [
+        dict(zip(predictions.keys(), v)) for v in zip(*predictions.values())
+    ]
+    return [
+        PredictionResult(x, y) for x, y in zip(batch, predictions_per_tensor)
+    ]
+  return [PredictionResult(x, y) for x, y in zip(batch, predictions)]
+
+
+def default_numpy_inference_fn(
+    inference_session: ort.InferenceSession,
+    batch: Sequence[numpy.ndarray],
+    inference_args: Optional[Dict[str, Any]] = None) -> Any:
+  ort_inputs = {inference_session.get_inputs()[0].name: numpy.stack(batch, axis=0)}
+  ort_outs = inference_session.run(None, ort_inputs)
+  return ort_outs
+
+
+class OnnxModelHandler(ModelHandler[numpy.ndarray,
+                                    PredictionResult,
+                                    ort.InferenceSession]):
+  def __init__(
+      self,
+      model_uri: str,
+      *,
+      inference_fn: NumpyInferenceFn = default_numpy_inference_fn):
+    """ Implementation of the ModelHandler interface for onnx
+    using numpy arrays as input.
+
+    Example Usage::
+
+      pcoll | RunInference(OnnxModelHandler(model_uri="my_uri"))
+
+    Args:
+      model_uri: The URI to where the model is saved.
+      inference_fn: The inference function to use.
+        default=default_numpy_inference_fn
+    """
+    self._model_uri = model_uri
+    self._model_inference_fn = inference_fn
+
+  def load_model(self) -> ort.InferenceSession:
+    """Loads and initializes an onnx inference session for processing."""
+    return _load_model(self._model_uri)
+
+  def run_inference(
+      self,
+      batch: Sequence[numpy.ndarray],
+      inference_session: ort.InferenceSession,
+      inference_args: Optional[Dict[str, Any]] = None
+  ) -> Iterable[PredictionResult]:
+    """Runs inferences on a batch of numpy arrays.
+
+    Args:
+      batch: A sequence of examples as numpy arrays. They should
+        be single examples.
+      inference_session: An onnx inference session. Must be runnable with input x where x is sequence of numpy array
+      inference_args: Any additional arguments for an inference.
+
+    Returns:
+      An Iterable of type PredictionResult.
+    """
+    predictions = self._model_inference_fn(inference_session, batch, inference_args)[0]
+
+    return _convert_to_result(batch, predictions)
+
+  def get_num_bytes(self, batch: Sequence[numpy.ndarray]) -> int:
+    """
+    Returns:
+      The number of bytes of data for a batch.
+    """
+    return sum(sys.getsizeof(element) for element in batch)
```

Comment:
Can we follow this for the num_bytes ?https://github.com/apache/beam/blob/95e53916b6c9de6052ce8fc8409a53bad3d9d517/sdks/python/apache_beam/ml/inference/tensorrt_inference.py#L307